hazelcast/src/main/java/com/hazelcast/core/IAtomicReference.java

```diff
@@ -122,4 +142,101 @@
      * @throws IllegalArgumentException if function is null.
      */
     <R> R apply(IFunction<E, R> function);
+
+    /**
+     * Atomically sets the value to the given updated value
+     * only if the current value {@code ==} the expected value.
+     *
+     * @param expect the expected value
+     * @param update the new value
+     * @return true if successful; or false if the actual value
+     *         was not equal to the expected value.
+     */
+    ICompletableFuture<Boolean> compareAndSetAsync(E expect, E update);
+
+    /**
+     * Gets the current value.
+     *
+     * @return the current value
+     */
+    ICompletableFuture<E> getAsync();
+
+    /**
+     * Atomically sets the given value.
+     *
+     * @param newValue the new value
+     */
+    ICompletableFuture<Void> setAsync(E newValue);
+
+    /**
+     * Gets the value and sets the new value.
+     *
+     * @param newValue the new value.
+     * @return the old value.
+     */
+    ICompletableFuture<E> getAndSetAsync(E newValue);
+
+    /**
+     * Sets and gets the value.
+     *
+     * @param update the new value
+     * @return  the new value
+     * @deprecated will be removed from Hazelcast 3.4 since it doesn't really serve a purpose.
```

Comment:
Agreed. Should we just remove it from this PR so it doesn't make it to the public API?